tests/Unit/GuzzleFileFetcherTest.php

```diff
@@ -127,4 +166,25 @@ public function testCreateFromUri(): void
         $this->expectExceptionMessage('Repo base URI must be HTTPS: http://example.com');
         GuzzleFileFetcher::createFromUri('http://example.com');
     }
+
+    /**
+     * Gets a fetcher instance that will throw a specified exceptoin.
+     *
+     * @param integer $statusCode
+     *   The response status code.
+     * @param string $exceptionClass
+     *   The expected exception class that will be thrown.
+     * @param integer|null $exceptionCode
+     *   (optional) The expected exception code. Defaults to the status code.
+     *
+     * @return GuzzleFileFetcher
+     *   The fetcher that wll producer an exception.
+     */
+    protected function getFetcherWithError(int $statusCode, string $exceptionClass, ?int $exceptionCode): GuzzleFileFetcher
+    {
+        $this->mockHandler->append(new Response($statusCode, [], $this->testContent));
+        $this->expectException($exceptionClass);
```

Comment:
Not sure how I feel about calling $this->expectException() and $this->expectExceptionCode() in a helper method; it obfuscates the expectations of the test. Not a big deal, but feels a bit "smelly" to me.

Reply:
yeah. I just removed this helper function. Agreed these are better in caller methods and then the new method really isn't necessary